### PR TITLE
Update KubeVirt platform how-to documentation

### PR DESCRIPTION
**Pull request by davidvossel**
This PR updates the KubeVirt platform's How-to documentation to reflect all the most recent improvements, changes, and recommendations.

### CODE REVIEW

File: docs/content/how-to/kubevirt/create-kubevirt-cluster.md [hunk 1]
```diff
@@ -4,228 +4,226 @@ Install a nested OCP cluster running on VMs within a management OCP cluster
...
-       ```shell
-       go get -u github.com/openshift/hypershift@latest
        ```
+       oc patch ingresscontroller -n openshift-ingress-operator default --type=json -p '[{ "op": "add", "path": "/spec/routeAdmission", "value": {wildcardPolicy: "WildcardsAllowed"}}]'
+       ```
+* The management OCP cluster must have Openshift Virtualization installed on it. [Instructions for installing Openshift Virtualization](https://docs.openshift.com/container-platform/4.12/virt/about-virt.html?extIdCarryOver=true&sc_cid=7013a0000026OSTAA2)
+* The Management OCP cluster must be configured with OVNKubernetes as the default pod network CNI.
+* The Management OCP cluster must have LoadBalancer service support. [Instructions for installing MetalLB](https://docs.openshift.com/container-platform/4.12/networking/metallb/metallb-operator-install.html)
+* The management OCP cluster must have default storage class. [Storage Configuration Documentation](https://docs.openshift.com/container-platform/4.12/post_installation_configuration/storage-configuration.html)
+    Below is an example of how to annotate a storageclass as the default storageclass.
```
Comment (gqlo) [inline docs/content/how-to/kubevirt/create-kubevirt-cluster.md:17]: The example given is to annotate one of the storage classes managed by ODF. Perhaps we wanna emphasize `ODF storageclass` . Here is the [installation guide](https://access.redhat.com/documentation/en-us/red_hat_openshift_data_foundation/4.12/html/deploying_openshift_data_foundation_using_bare_metal_infrastructure/deploy-using-local-storage-devices-bm) if you think it's necessary to include it somewhere.

File: docs/content/how-to/kubevirt/create-kubevirt-cluster.md [hunk 2]
```diff
@@ -4,228 +4,226 @@ Install a nested OCP cluster running on VMs within a management OCP cluster
...
+
+```shell
+podman run --rm --privileged -it -v \
+$PWD:/output docker.io/library/golang:1.18 /bin/bash -c \
+'git clone https://github.com/openshift/hypershift.git && \
+cd hypershift/ && \
+make hypershift && \
+mv bin/hypershift /output/hypershift'
+```
+
```
Comment (gqlo) [inline docs/content/how-to/kubevirt/create-kubevirt-cluster.md:40]: Do we want to ask the user to place the hypershift binary somewhere included in the `$PATH`? , something like: 
`sudo mv $PWD/hypershift /usr/local/bin`

File: docs/content/how-to/kubevirt/create-kubevirt-cluster.md [hunk 3]
```diff
@@ -4,228 +4,226 @@ Install a nested OCP cluster running on VMs within a management OCP cluster
...
+apiVersion: metallb.io/v1beta1
+kind: MetalLB
+metadata:
+  name: metallb
+  namespace: metallb-system
+EOF
 ```
-cat << EOF > apps-ingress-service.yaml
-apiVersion: v1
-kind: Service
```
Comment (gqlo) [inline docs/content/how-to/kubevirt/create-kubevirt-cluster.md:140]: Oh this part will be covered by this [PR](https://github.com/openshift/hypershift/pull/1787) ?  As long as user are building the binary from main branch, we could use the Default Ingress Passthrough feature, right?

**Review by gqlo — COMMENTED** [review]
Just some minor comments. Otherwise, everything looks good to me.

File: docs/content/how-to/kubevirt/create-kubevirt-cluster.md (hunk 3 above)
Comment (davidvossel) [reply docs/content/how-to/kubevirt/create-kubevirt-cluster.md:140]: yes. This also happens to work successfully in MCE 2.2 with any ocp guest cluster that is 4.12.2 or greater. Hypershift got a backport to 4.12 that included this basedoman passthrough work.

File: docs/content/how-to/kubevirt/create-kubevirt-cluster.md (hunk 2 above)
Comment (davidvossel) [reply docs/content/how-to/kubevirt/create-kubevirt-cluster.md:43]: fixed

File: docs/content/how-to/kubevirt/create-kubevirt-cluster.md (hunk 1 above)
Comment (davidvossel) [reply docs/content/how-to/kubevirt/create-kubevirt-cluster.md:17]: I don't think I'm ready to talk about ODF quite yet. We'll need a RWX csi driver once we support live migration, but we're about 5 months away from that being a reality.

File: docs/content/how-to/kubevirt/create-kubevirt-cluster.md [hunk 4]
```diff
@@ -4,228 +4,226 @@ Install a nested OCP cluster running on VMs within a management OCP cluster
...
+* Admin access to an OpenShift cluster (version 4.12+) specified by the `KUBECONFIG` environment variable.
+* The management OCP cluster must have wildcard dns routes enabled.
 
-    Install it using Go 1.18+:
-       ```shell
-       go get -u github.com/openshift/hypershift@latest
        ```
+       oc patch ingresscontroller -n openshift-ingress-operator default --type=json -p '[{ "op": "add", "path": "/spec/routeAdmission", "value": {wildcardPolicy: "WildcardsAllowed"}}]'
+       ```
+* The management OCP cluster must have Openshift Virtualization installed on it. [Instructions for installing Openshift Virtualization](https://docs.openshift.com/container-platform/4.12/virt/about-virt.html?extIdCarryOver=true&sc_cid=7013a0000026OSTAA2)
```
[the remark this answers is not in the document]
Comment (davidvossel) [reply docs/content/how-to/kubevirt/create-kubevirt-cluster.md]: yep, fixed

**Review by gqlo — APPROVED** [review]
Thanks @davidvossel Everything looks good to me

**Pull request merged**